core/src/price_estimation/threaded_price_source.rs

```diff
@@ -46,6 +47,14 @@ impl ThreadedPriceSource {
     }
 }
 
+async fn update_prices<T: PriceSource>(
+    price_source: &T,
+    token_info_fetching: &Arc<dyn TokenInfoFetching>,
```

Comment:
In general `&Arc<dyn Trait>` should be `&dyn Trait` because either way it's just a reference but the second signature is more flexible. It would be called by turning `&token_info_fetcher` into `token_info_fetcher.as_ref()`.
Don't need to change it here because this is really only meant to be called from that one place.

Reply:
Good point! Will change it to make it a habit of not using &Arc

Reply:
Good catch! One "valid" use of `&Arc` is if you intend to clone it. Although, in that case its probably better to pass in an owned value `Arc` instead, and have the caller clone it if it needs to.